src/libs/copy_history.c

```diff
@@ -272,14 +274,14 @@ static void discard_button_clicked(GtkWidget *widget, gpointer user_data)
 
   if(res == GTK_RESPONSE_YES)
   {
+    GList *imgs_copy = g_list_copy((GList *)imgs);
```

Comment:
why is this copy needed? Not needed in dt_history_delete_on_list and it is copied again in next instruction (dt_collection_update_query).

Reply:
Missed one.... Thanks for catching that.